### PR TITLE
chore(TS): Remove stateful mixin

### PR DESCRIPTION
**Pull request by asturur**
## Motivation

The Stateful mixin had 2 defects.
1) it tried to detect change on objects on the fly ( complex and not good for performances )
2) It had some weird comparing code we don't want to maintain

The direction

## Description
- remove stateful mixin and its tests

## Changes
- renamed _dimensionAffectingProps to textLayoutProperties, static.
- changes to textLayoutProperties require the use of set, exactly as cache properties

### CODE REVIEW

File: src/shapes/IText/IText.ts [hunk 1]
```diff
@@ -149,6 +150,8 @@ export class IText<
    */
   declare caching: boolean;
 
+  static textLayoutProperties = Text.textLayoutProperties;
```
Comment (ShaMan123) [inline src/shapes/IText/IText.ts:153]: no need for this
static props are inherited

File: src/shapes/Text/Text.ts [hunk 2]
```diff
@@ -79,6 +79,22 @@ const additionalProps = [
   'pathAlign',
 ] as const;
 
+const textLayoutProperties: string[] = [
```
Comment (ShaMan123) [inline src/shapes/Text/Text.ts:82]: why not define it directly on the class?

**Review by ShaMan123 — COMMENTED** [review]
Looks great
Can stateProps and cacheProps become static as  well??

File: src/shapes/Text/Text.ts [hunk 3]
```diff
@@ -1690,11 +1703,11 @@ export class Text<
           this.setPathInfo();
         }
         needsDims =
-          needsDims || this._dimensionAffectingProps.indexOf(_key) !== -1;
+          needsDims || this.constructor.textLayoutProperties.includes(_key);
         isAddingPath = isAddingPath || _key === 'path';
       }
     } else {
-      needsDims = this._dimensionAffectingProps.indexOf(key) !== -1;
+      needsDims = this.constructor.textLayoutProperties.includes(key);
```
Comment (asturur) [inline src/shapes/Text/Text.ts:1710]: This thing here remind me of te issue we have with side effects.
sometimes those fit in _set, but if those are slightly expensive we want to run them only once for set.
But this bring us to handle in set both single and multi key use case.
@ShaMan123  should check your side effects pr solves this issue with 10 lines of code or find a valid way to remove those with a little footprint.

File: src/shapes/IText/IText.ts (hunk 1 above)
Comment (ShaMan123) [reply src/shapes/IText/IText.ts:153]: I see #8662

File: src/shapes/Text/Text.ts (hunk 3 above)
Comment (ShaMan123) [reply src/shapes/Text/Text.ts:1710]: don't think it is 10 lines but it solves this bad design

File: src/shapes/Text/Text.ts (hunk 2 above)
Comment (asturur) [reply src/shapes/Text/Text.ts:82]: i like the single line of the claration

File: src/shapes/IText/IText.ts (hunk 1 above)
Comment (asturur) [reply src/shapes/IText/IText.ts:153]: are they? my understand is that they are inherited only in static methods.
If is not needed even better.

**Comment by ShaMan123** [comment]
Another thought
We might want to get rid of static fields because they compile to a side effect and a method doesn't or is it a matter of fine tuning rollup/browserlist?

File: src/shapes/IText/IText.ts (hunk 1 above)
Comment (ShaMan123) [reply src/shapes/IText/IText.ts:153]: image

```js
class A {
 static m=5
}
class B extends A {}
console.log(B.m)
```

**Comment by asturur** [comment]
they compile to a sideEffect you mean they become
``` Object.prop = value ``` ?
I m going to check what the output is like.

Seems like classes aren't ready if i have to create a getter for a prop because they are handled differently.
It also leaves me with the issue of creating a setter too.

**Comment by asturur** [comment]
yes so they become side effects.
When we get to code tree shaking we have to decide.
It could also be that we leave the lib unbundled for who wants to optimize the bundle and imports the single files.
The other will use a large premade bundle built with rollup.

I m kind of tired to spinning in circle to js/ts issues
This migration definitely took a toll on me.

**Comment by asturur** [comment]
This is ready to go, not tackling side effects now, the scope is removing stateful.

**Review by ShaMan123 — APPROVED** [review]
Yes
The maintenance is a really unpleasant part.
I looked at the links of tree shaking in rollup
https://rollupjs.org/configuration-options/#treeshake-modulesideeffects

We can expose a rollup config for consumers or whatever
Never mind that lets move fabric forward
It has taken a toll on me too

**Pull request merged**